File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = set(['a','e','i','o','u'])

        maxVowels = 0

        for char in s[0:k]:
            if char in vowels:
                maxVowels += 1

        l, r, currVowels = 0, k, maxVowels

        while r < len(s):
            if s[l] in vowels:
                currVowels -= 1

            l += 1

            if s[r] in vowels:
                currVowels += 1

            r += 1

            maxVowels = max(currVowels, maxVowels)

        return maxVowels 

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
static const bool Booster = [](){
    std::ios_base::sync_with_stdio(false), std::cout.tie(nullptr), std::cin.tie(nullptr);
    return true;
}();
class Solution {
public:
    const int maxVowels(const string s, const int k) {
        unordered_set<char> vowelSet {'a', 'e', 'i', 'o', 'u'};
        int vowelCount = 0;
        for (int i = 0; i < k && i < s.size(); ++i) {
            if (vowelSet.contains(s[i])) ++vowelCount;
        }
        int l = 0, r = k, result = vowelCount;
        while (r < s.size()) {
            if (vowelSet.contains(s[l])) --vowelCount;
            if (vowelSet.contains(s[r])) ++vowelCount;
            result = max(result, vowelCount);
            ++l;
            ++r;
        }
        return result;
    }
};